// @trpc/server/dist/trpc-server.cjs.prod.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var resolveHTTPResponse = require('./resolveHTTPResponse-9812e803.cjs.prod.js');
var transformTRPCResponse = require('./transformTRPCResponse-077b7539.cjs.prod.js');
var router = require('./router-aaecbe82.cjs.prod.js');
var subscription = require('./subscription-54d50ff1.cjs.prod.js');
var adapters_standalone_dist_trpcServerAdaptersStandalone = require('../adapters/standalone/dist/trpc-server-adapters-standalone.cjs.prod.js');
var nodeHTTPRequestHandler = require('./nodeHTTPRequestHandler-21707788.cjs.prod.js');
require('./codes-aff770a3.cjs.prod.js');
require('events');
require('http');
require('url');

// this has been moved to separate package

/**
 * @deprecated use `createHTTPServer` from `@trpc/server/adapters/standalone`
 */
const createHttpServer = adapters_standalone_dist_trpcServerAdaptersStandalone.createHTTPServer;
/**
 * @deprecated use `createHTTPHandler` from `@trpc/server/adapters/standalone`
 */

const createHttpHandler = adapters_standalone_dist_trpcServerAdaptersStandalone.createHTTPHandler;
/**
 * @deprecated use `CreateHTTPHandlerOptions` from `@trpc/server/adapters/standalone`
 */

/**
 * @deprecated use `nodeHTTPRequestHandler` from `@trpc/server/adapters/node-http`
 */

const requestHandler = nodeHTTPRequestHandler.nodeHTTPRequestHandler;
/**
 * @deprecated use `NodeHTTPCreateContextFn` from `@trpc/server/adapters/node-http`
 */

exports.assertNotBrowser = resolveHTTPResponse.assertNotBrowser;
exports.resolveHTTPResponse = resolveHTTPResponse.resolveHTTPResponse;
exports.TRPCError = transformTRPCResponse.TRPCError;
exports.callProcedure = transformTRPCResponse.callProcedure;
exports.getErrorFromUnknown = transformTRPCResponse.getErrorFromUnknown;
exports.getMessageFromUnkownError = transformTRPCResponse.getMessageFromUnkownError;
exports.transformTRPCResponse = transformTRPCResponse.transformTRPCResponse;
exports.LegacyRouter = router.LegacyRouter;
exports.Router = router.LegacyRouter;
exports.router = router.router;
exports.Subscription = subscription.Subscription;
exports.subscriptionPullFactory = subscription.subscriptionPullFactory;
exports.createHttpHandler = createHttpHandler;
exports.createHttpServer = createHttpServer;
exports.requestHandler = requestHandler;
